src/main/java/com/google/cloud/spanner/pgadapter/wireoutput/ErrorResponse.java

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.cloud.spanner.pgadapter.wireoutput;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.text.MessageFormat;
+
+/**
+ * Sends error information back to client.
+ */
+public class ErrorResponse extends WireOutput {
+
+  private static final byte[] SEVERITY = "ERROR".getBytes(UTF8);
+  private static final byte[] SQL_STATE = "XX000".getBytes(UTF8);
+  private static final int HEADER_LENGTH = 4;
+  private static final int FIELD_IDENTIFIER_LENGTH = 1;
+  private static final int NULL_TERMINATOR_LENGTH = 1;
+
+  private static final byte CODE_FLAG = 'C';
+  private static final byte MESSAGE_FLAG = 'M';
+  private static final byte SEVERITY_FLAG = 'S';
+  private static final byte NULL_TERMINATOR = 0;
+
+
+  private final byte[] errorMessage;
+  private final byte[] errorState;
+
+  public ErrorResponse(DataOutputStream output, Exception e, State errorState) {
```

Comment:
we might need to pull the error state from the exception itself.  Ok to do that in a subsequent PR (add TODO)

Reply:
Done